pkg/activator/net/revision_backends.go

```diff
@@ -317,7 +317,9 @@ func (rw *revisionWatcher) checkDests(curDests, prevDests dests) {
 		rw.logger.Debugf("Done probing, got %d healthy pods", len(hs))
 		if !noop || len(reprobe) > 0 {
 			rw.healthyPods = hs
-			rw.sendUpdate("" /*clusterIP*/, hs)
+			// Copy the healthy pods set before sending the update to avoid
+			// concurrent modifications affecting the throttler.
+			rw.sendUpdate("" /*clusterIP*/, sets.NewString(hs.List()...))
```

Comment:
`List()` is kind of an expensive operation, we should instead use `UnsortedList()` which at least drops the need to sort the strings as well.
Diggin a bit deeper into the implementations, we could actually also use `hs.Union(nil)` I think to even drop the intermediate string slice and directly construct a new set.

Reply:
yeah I did think about using something like Union but it seemed a bit unclear to rely on the side effect of it copying, actually though there's a huge comment here explaining the copy so 🤷 one sec, I'll update

Reply:
pushed

Reply:
FWIW this annoyed me so I sent https://github.com/kubernetes/gengo/pull/213 to hopefully make this better for others in the future :joy:

Reply:
heh, nice!